server/src/main/java/io/deephaven/server/session/SessionState.java

```diff
@@ -402,12 +402,11 @@ public void addOnCloseCallback(final Closeable onClose) {
      *          already been invoked or will be invoked by the SessionState.
      */
     public boolean removeOnCloseCallback(final Closeable onClose) {
+        if (isExpired()) {
```

Comment:
Good change. Should we make `addOnCloseCallback` do a double check of `isExpired()`? That is, should we check it before and after the lock?